File: activity/templates/app-java/src/main/java/ui/_Fragment.java
package <%= appPackage %>.ui.<%= activityPackageName %>;

import android.support.v4.app.Fragment;

<% if (componentType == 'createNew') { %>
import <%= appPackage %>.di.components.<%= activityName %>Component;
<% } else if (componentType == 'useApplication') { %>
import <%= appPackage %>.application.App;
<% } %>
<% if (nucleus == true) { %>import nucleus.factory.PresenterFactory; <% } %>

import javax.inject.Inject;

import <%= appPackage %>.ui.base.BaseFragment;
import <%= appPackage %>.ui.base.EmptyPresenter;
import <%= appPackage %>.R;


public class <%= activityName %>Fragment extends BaseFragment<EmptyPresenter>  {

    @Inject
    EmptyPresenter emptyPresenter;

    @Override
    protected void inject() {
        <% if (componentType == 'createNew') { %>getComponent(<%= activityName %>Component.class).inject(this);<% } else { %>App.graph.inject(this);<% } %>
    }

    <% if (nucleus == true) { %>
    public PresenterFactory<EmptyPresenter> getPresenterFactory() {
        return () -> emptyPresenter;
    }<% } %>
    <% if (nucleus == true) { %>
    public PresenterFactory<EmptyPresenter> getPresenterFactory() {
        return () -> emptyPresenter;
    }<% } else { %>
    @Override
    protected EmptyPresenter getPresenter() {
        return emptyPresenter;
    }<% } %>


    @Override
    protected int getLayoutResource() {
        return R.layout.fragment_<%= underscoreActivityName %>;
    }

    public static Fragment newInstance() {
        return new <%= activityName %>Fragment();
    }
}

File: fragment/templates/app-kotlin/src/main/java/_Fragment.kt
package <%= appPackage %>.ui.<%= fragmentPackageName %>

import javax.inject.Inject

<% if (componentType == 'createNewSub') { %>import <%= appPackage %>.di.components.<%= useExistingComponentName %>Component<% } else if (componentType == 'createNew') { %>import <%= appPackage %>.di.components.<%= activityName %>Component<% } else if (componentType == 'useApplication' || componentType == 'useUserComponent') { %>import <%= appPackage %>.application.App<% } else { %>import <%= appPackage %>.application.App
import <%= appPackage %>.di.components.<%= useExistingComponentName %>Component<% } %>
<% if (usePresenter == false) { %>import <%= appPackage %>.ui.base.EmptyPresenter<% } %>
<% if (usePresenter == false) { %>import <%= appPackage %>.ui.base.EmptyView<% } %>
import <%= appPackage %>.ui.base.BaseFragment
import <%= appPackage %>.R

class <%= fragmentName %>Fragment : BaseFragment<<% if (usePresenter) { %><%= fragmentName %><% } else { %>Empty<% } %>Presenter>(), <% if (usePresenter) { %> <%= fragmentName %><% } else { %>Empty<% } %>View {

    @Inject
    lateinit var <% if (usePresenter) { %><%= fragmentName.charAt(0).toLowerCase()+fragmentName.slice(1) %><% } else { %>empty<% } %>Presenter : <% if (usePresenter) { %><%= fragmentName %><% } else { %>Empty<% } %>Presenter

    override fun onResume() {
        super.onResume()
        <% if (usePresenter) { %><%= fragmentName.charAt(0).toLowerCase()+fragmentName.slice(1) %><% } else { %>empty<% } %>Presenter.takeView(this)
    }

    override fun onPause() {
        super.onPause()
        <% if (usePresenter) { %><%= fragmentName.charAt(0).toLowerCase()+fragmentName.slice(1) %><% } else { %>empty<% } %>Presenter.dropView()
    }

    override fun inject() {
        <% if (componentType == 'createNewSub') { %>context?.let { getComponent(<%= useExistingComponentName %>Component::class.java).inject(this) }<% } else if (componentType == 'createNew') { %>context?.let { getComponent(<%= activityName %>Component::class.java).inject(this) }<% } else if (componentType == 'useApplication') { %>context?.let { App.get(it).getComponent().inject(this) }<% } else if (componentType == 'useUserComponent') { %>context?.let { App.get(it).getUserComponent()?.inject(this) }<% } else { %>context?.let { App.get(context).get<%= useExistingComponentName.replace('Application', '') %>Component().inject(this) }<% } %>
    }

    override fun getPresenter(): <% if (usePresenter) { %><%= fragmentName %><% } else { %>Empty<% } %>Presenter = <% if (usePresenter) { %><%= fragmentName.charAt(0).toLowerCase()+fragmentName.slice(1) %><% } else { %>empty<% } %>Presenter

    override fun getLayoutResource(): Int = R.layout.fragment_<%= underscoreFragmentName %>

}
